### PR TITLE
Remove arbitrary or unnecessary content from make toolchain

### PR DESCRIPTION
**Pull request by solvingj**
From conversation on slack with @memsharded :

> at the moment, most of the toolchain is unused, right?

> maybe we want to start super minimalistic instead?

> something like "things not necessary to have the tests passing".

This implementation removes everything not necessary to have tests passing on linux and windows (mingw32). I think it is minimalistic, but @memsharded please advise if anything else should be changed. 

Changelog: Fix: Simplified MakeToolchain to remove things that were not checked by tests or unused.
Docs: omit

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

#TAGS: slow

### CODE REVIEW

**Comment by solvingj** [comment]
@memsharded please review (I don't have permission to add a reviewer.)

File: conans/client/toolchain/make.py [hunk 1]
```diff
@@ -22,172 +16,26 @@ class MakeToolchain(object):
...
-        {%- endif %}
-
             # Recipe-Defined Variables
         {%- for it, value in variables.items() %}
             {{ it }} = {{ value }}
         {%- endfor %}
 
-            # Automatic Conan pre-processor definition: build_type_define
-        {%- if build_type_define %}
-            CONAN_TC_CPPFLAGS += -D{{build_type_define}}
```
Comment (memsharded) [inline conans/client/toolchain/make.py]: But this was being used, it was already part of the build

File: conans/client/toolchain/make.py [hunk 2]
```diff
@@ -22,172 +16,26 @@ class MakeToolchain(object):
...
-        {%- endif %}
-
-            # Automatic Conan pre-processor definition: glibcxx_define
-        {%- if glibcxx_define %}
-            CONAN_TC_CPPFLAGS += -D{{glibcxx_define}}
```
Comment (memsharded) [inline conans/client/toolchain/make.py]: Same: this was part of the build

File: conans/client/toolchain/make.py [hunk 3]
```diff
@@ -22,172 +16,26 @@ class MakeToolchain(object):
         ifndef CONAN_TOOLCHAIN_INCLUDED
             CONAN_TOOLCHAIN_INCLUDED = TRUE
 
-            # Automatic Conan Toolchain Variables
-        {%- if build_type %}
-            CONAN_TC_BUILD_TYPE = {{build_type}}
```
Comment (memsharded) [inline conans/client/toolchain/make.py:28]: This was not part of the build

**Review by memsharded — CHANGES_REQUESTED** [review]
I am afraid there was some misunderstanding, it seems you went too far. 
I meant, all the vars that are not being used at all in the build, like CONAN_TC_ARCH_HOST.

Other vars, they were actually involved in the build, and they were being injected in CONAN_TC_SETUP, right? Then we have at least some evidence that they are not completely broken.

File: conans/client/toolchain/make.py (hunk 2 above)
Comment (solvingj) [reply conans/client/toolchain/make.py]: restored

File: conans/client/toolchain/make.py (hunk 3 above)
Comment (solvingj) [reply conans/client/toolchain/make.py:28]: no-op

File: conans/client/toolchain/make.py (hunk 1 above)
Comment (solvingj) [reply conans/client/toolchain/make.py]: restored

**Comment by solvingj** [comment]
I've restored the specific preprocessor definitions requested.  Please name or highlight any other specific flags or definitions which you would like restored.  I think it's best for you to choose the list explicitly in this case.

**Comment by solvingj** [comment]
@memsharded Here are four items related to settings which were removed.  Please indicate any of these which should be restored.
```
libcxx_flag 
cppstd_flag
build_type_flags
arch_flag
```

**Comment by memsharded** [comment]
> libcxx_flag 
cppstd_flag
build_type_flags
arch_flag

These should be restored, making sure they are actually being used in the compilation, checking the binary or introducing architecture or cppstd specific code, it depends.
Maybe it can be delayed until users feedback, to learn actual user needs and patterns. For example, is it expected that Conan defines "Debug" or "Release" CXXFLAGS? Maybe not, that would be part of users Makefiles, right? Maybe it is an opt-in for users that want us to give the utility?

**Comment by memsharded** [comment]
I have merged it, lets build step by step on the minimalistic toolchain

**Pull request merged**